web/components/roadmap/MiniRoadmap.test.tsx

```diff
@@ -91,18 +95,77 @@ describe("<MiniRoadmap />", () => {
       ],
     });
 
-    useMockUserData({
+    const subject = renderMiniRoadMap(
+      "task3",
+      generateUserData({
+        preferences: generatePreferences({
+          roadmapOpenSections: ["PLAN", "START"],
+        }),
+      })
+    );
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  describe("User Step State Preferences", () => {
+    const userData = generateUserData({
       preferences: generatePreferences({
         roadmapOpenSections: ["PLAN", "START"],
+        roadmapOpenSteps: [2],
       }),
     });
+    beforeEach(() => {
+      useMockRoadmap({
+        steps: [
+          generateStep({
+            name: "step1",
+            step_number: 1,
+            section: "PLAN",
+            tasks: [generateTask({ name: "task1", id: "task1" })],
+          }),
+          generateStep({
+            name: "step2",
+            step_number: 2,
+            section: "START",
+            tasks: [generateTask({ name: "task2", id: "task2" })],
+          }),
+        ],
+      });
+    });
 
-    const subject = renderMiniRoadMap("task3");
+    it("displays steps based on userData preferences", async () => {
+      const subject = renderMiniRoadMap("task1", userData);
+      expect(subject.getByText("task2")).toBeInTheDocument();
+    });
 
-    const sectionStart = subject.getByTestId("section-start");
-    const sectionPlan = subject.getByTestId("section-plan");
+    it("does not displays steps if not in userData preferences", async () => {
```

Comment:
"displays" -> "display"

Reply:
and not to nitpick for these test names but technically it _does_ display the step even if it's not in preferences. it just doesn't display the tasks within the step